arduino-ide-extension/src/node/clang-formatter.ts

```diff
@@ -147,189 +147,10 @@ function styleJson({
   UseTab,
 }: ClangFormatOptions): Record<string, unknown> {
   // Source: https://github.com/arduino/tooling-project-assets/tree/main/other/clang-format-configuration
+  const defaultConfig = require('./default-formatter-config.json'); // 1. require the JSON
```

Comment:
This line should be changed to, it's my fault. I found it only when verifying the behavior:

```ts
const defaultConfig = require('../../src/node/default-formatter-config.json');
```

Explanation:

We write TS code that will be transpiled to JS code. IDE2 does not run TS code, only JS code. The sources (TS files, JSON, .etc) are in `src`, the transpiled JS code is in `lib`.

The TS sources and the JSON config are here:
```
arduino-ide-extension/src/node/clang-formatter.ts
arduino-ide-extension/src/node/default-formatter-config.json
```

The TS code will be JS, and the module location will be these:
```
arduino-ide-extension/lib/node/clang-formatter.d.ts
arduino-ide-extension/lib/node/clang-formatter.d.ts.map
arduino-ide-extension/lib/node/clang-formatter.js
```

Notice the `src` and `lib` differences in the path. JSON files will remain in `src` at runtime, JS files will be in `lib` folder. To be able to `require` the JSON file from the transpiled JS, the `require` path must be changed from `./default-formatter-config.json` to `../../src/node/default-formatter-config.json`. If this is not clear, please let me know.

Also, please remove the `// 1. require the JSON` and the `// 2. override the default values with the user-defined ones` comments. I added them for explanation purposes.

Thank you!

Reply:
>This line should be changed

Thanks. I have done it here: https://github.com/arduino/arduino-ide/pull/1324/commits/becb9013da4f915aec3e138bb10a14903698b901

>JSON files will remain in src at runtime, JS files will be in lib folder.

I saw that when I went to test the build (I think I somehow ended up still old source code when I ran it from source).

I was going down the path mentioned here:

https://stackoverflow.com/a/53629381/7059512

(adding something like `"src/node/*.json"` to the `include` field of `tsconfig.json`)

>please remove the // 1. require the JSON and the // 2. override the default values with the user-defined ones comments.

OK, done here: https://github.com/arduino/arduino-ide/pull/1324/commits/aff356078cfa066215b445cc925d918876a6d5e2